File: 1 Python/Codeforces-1560B-WhosOpposite.py
nc = int(input())

for _ in range(nc):
    a,b,c = map(int, input().rstrip().split())
    diff = abs(b-a)
    ceiling = 2*diff
    # if we already have an invalid state, we discard
    if a>ceiling or b>ceiling or c>ceiling or diff==1:
        ans = -1
    else:
        if c+diff>ceiling:
            ans = c-diff
        else:
            ans = c+diff
    print(ans)

File: 2 CPP/Codeforces-1560B-WhosOpposite.cpp
#include <bits/stdc++.h>
using namespace std;
using l2 = long;

void solve(){
    l2 a,b,c;
    cin >> a >> b >> c;
    l2 diff = abs(b-a);
    l2 ceiling = 2*diff;
    l2 ans;
    if(a>ceiling || b> ceiling || c>ceiling || diff==1 ){
        ans = -1;
    }
    else{
        ans = ((c+diff)>ceiling) ? c-diff : c+diff;
    }
    cout << ans << "\n";
}

int main(){
    l2 tc;
    cin >> tc;
    while(tc--){
        solve();
    }
    return 0;
}
